### ISSUE TITLE
"Manage Account" opens accounts.firefox.com in a page that requests your email

### ISSUE THREAD

**Issue by mhammond** [mozilla-mobile/firefox-ios#12069]
## Steps to reproduce
* Sign in to Sync
* Select "manage account"

### Expected behavior
* You are taken to the "manage account" page for the account signed in to your device

### Actual behavior
* You are taken to an "Enter your email" page with a "Sign up or sign in" link.

I'm not that familiar with how the iOS account integration works, but it may be difficult to avoid prompting for a password, but it certainly is possible to avoid asking for the email address (ie, iOS can provide the email address of the signed in user in the URL)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-5035)

**Comment by mhammond** [comment]
And worse, if you change you password on this page, you are forced to sign in again in the browser - ie, changing your password on this form leaves Firefox disconnected in the browser.

**Comment by mhammond** [comment]
It looks like at least one of those signins was in safari - my "manage devices" now shows this. Note that this Firefox is not my default browser.

![image](https://user-images.githubusercontent.com/83924/194002666-ff0e46f4-aaa5-41d5-b80a-3e4094d3e7b4.png)

**Comment by mhammond** [comment]
I can't reproduce this as reported, so I'll close it until I can.

**Comment by tarikeshaq** [comment]
This is not fixed yet - Your first time clicking on "Manage Account" in the setting menu still requests your email/password.

The problem here is more fundamental, the device stores a refresh token and not a session token, but to access that site you need a session token so you sign in again. The solution here probably involves coordination between the content server and the browser. See also https://bugzilla.mozilla.org/show_bug.cgi?id=1753785

**Comment by tarikeshaq** [comment]
> and not a session token

This may not be correct, but **if** the device is holding on to a session token, it's not setting it properly for the webview, I can quickly verify

**Comment by tarikeshaq** [comment]
Yeah, after we switched to oauth (and the Rust component) a while ago, devices no longer store a session token, they do when they migrate from the (very) old flow, but that's long enough ago that I can't imagine many devices still have it.

The Rust component does have API's to store it, and it also has an API to use a session token when a user resets their password... Maybe it's extendable but will still require a decent bit of work here and in the content server to pass a session token back (and of course I'm not sure that's what we want either soo 😅 )

**Comment by mhammond** [comment]
bad bot
